pipeline/utils/__init__.py

```diff
@@ -0,0 +1,62 @@
+import json
+import os
+from io import BufferedReader
+
+import requests
+
+
+class Dataset:
```

Comment:
so based on all the comments this should live in the core pipeline

Reply:
yes, if this will be reused across different pipeline scripts. I always prefer creating a named python module instead of putting things in `__init__.py`. It's way easier to read and discover code this way. So it could be `pipeline/common/dataset.py`